Changes to hardware.mk for simulation, small improvement to read_process_native

joaovroque opened the pull request:
- Change to hardware.mk to work with the multiple simulation options
- Small change to read_process_native to be able to read 1 word/clock-cycle (in a 0 read-latency scenario). Doesn't affect L2 architectures (it has a 1 clock read-latency).

jjts commented on hardware/hardware.mk:
```diff
@@ -15,4 +15,7 @@ CACHE_SRC_DIR:=$(CACHE_DIR)/hardware/src
-$(MEM_DIR)/sp_ram/iob_sp_mem.v
+$(MEM_DIR)/sp_ram_be/iob_sp_ram_be.v \
+$(MEM_DIR)/sp_ram/iob_sp_mem.v \
+$(AXI_MEM_DIR)/rtl/axi_ram.v \
```
AXI_MEM is only used for simulation. This file is only for implementation hardware. You should make your own simulation.mk file.

iob_sp_ram_be is not used in the system. It should not be uesed at all because Altera tools do not support it.

jjts replied:
please fix

joaovroque commented:
hardware.mk changed.
The sources for the simulation were placed in simulation.mk correctly.

jjts commented:
please answer question

joaovroque commented:
It is answered. The system.mk only has the sources for implementation.
simulation.mk has all  the additional sources for simulation (axi_ram.v, etc).
The commit wasn't updated because it was soft-reseted with the fixes.
The large amount of deleted lines (-97,400) were because the make clean wasn't working properly in the simulation folder, so it was cleaned.

joaovroque commented:
Changed the replacement policy testbench Makefile to be able to run it's testbench.
Implemented a much simpler version of Bit-pLRU replacement policy. Also uses less resources. May require more testings to be sure, but simulations showed a correct behaviour (even with the Random hits generation).
